numpy/core/src/multiarray/common.c

```diff
@@ -632,7 +632,17 @@ _IsWriteable(PyArrayObject *ap)
         ap = (PyArrayObject *)base;
         base = PyArray_BASE(ap);
 
+        if (PyArray_ISWRITEABLE(ap)) {
+            /*
+             * If any base is writeable, it must be OK to switch, note that
+             * bases are typically collapsed to always point to the most
+             * general one.
+             */
+            return NPY_TRUE;
+        }
+
         if (base == NULL || PyArray_CHKFLAGS(ap, NPY_ARRAY_OWNDATA)) {
+            /* there is no further base to test the writeable flag for */
             return (npy_bool) PyArray_ISWRITEABLE(ap);
```

Comment:
I think it would be clearer if this were `return False`, since at this point `ap` can never be writeable

Reply:
adopting